mpas_analysis/shared/climatology/climatology.py

```diff
@@ -608,17 +806,23 @@ def ds_to_weights(ds):
             for var in ds.data_vars:
                 weights[var] = ds[var].notnull()
         else:
-            raise TypeError('ds must be an instance of either xarray.Dataset or xarray.DataArray.')
+            raise TypeError('ds must be an instance of either xarray.Dataset '
+                            'or xarray.DataArray.')
 
         return weights
 
-    dsWeightedSum = (ds * ds.daysInMonth).sum(dim='Time', keep_attrs=True)
+    if maskVaries:
+        dsWeightedSum = (ds * ds.daysInMonth).sum(dim='Time', keep_attrs=True)
 
-    weights = ds_to_weights(ds)
+        weights = ds_to_weights(ds)
 
-    weightSum = (weights * ds.daysInMonth).sum(dim='Time')
+        weightSum = (weights * ds.daysInMonth).sum(dim='Time')
 
-    timeMean = dsWeightedSum / weightSum.where(weightSum > 0.)
+        timeMean = dsWeightedSum / weightSum.where(weightSum > 0.)
+    else:
+        days = ds.daysInMonth.sum(dim='Time')
+        dsWeightedSum = (ds * ds.daysInMonth).sum(dim='Time', keep_attrs=True)
+        timeMean = dsWeightedSum / days.where(days > 0.)
```

Comment:
@xylar, the slowdown is probably larger than a factor of two at least originally.  

This code still has a 2X slowdown.  Taking two sums does not alleviate the problem because two sums takes twice as long as a mean because two reduction operations need to be done.  The difference of mean vs accounting for the month time variation results in a difference of < 5% per year.  Maybe I'm missing something but I would suspect we have larger errors more worthy of our time.  When the rush is off if you could please clarify why this added precision is needed I would be grateful.

Thanks for adding more docstrings to clarify.